gems/laa_multi_step_forms/docs/SharedForms.md

```diff
@@ -0,0 +1,105 @@
+# Shared forms
+
+These are form objects that are generic enough the have been added into the gem
+and can be reused in specific situations
+
+## AddAnotherForm
+
+This form is designed to be used when it is possible to add multiple instance of
+an object, and the index page asks the question `Do you want to add another`.
+
+The form has a single YesNoAnswer attribute and does not modify the application
+when processed.
+
+When using this pattern:
+*  the controller will need to load any other data need for
+the view into a separate data variable, this should be done in both thge `edit`
+and `update` endpoints.
+* `additional_permitted_params` need to be set to `[:add_another]` otherwise
+the users selection will be ignored
+
+example:
+
+```ruby
+module Steps
+  class ItemsController < Steps::BaseStepController
+    def edit
+      @data_for_view = current_application.data_for_view
+      @form_object = AddAnotherForm.build(
+        current_application
+      )
+    end
+
+    def update
+      @data_for_view = current_application.data_for_view
+      update_and_advance(AddAnotherForm, as: :work_items)
+    end
+
+    private
+
+    def decision_tree_class
+      Decisions::SimpleDecisionTree
+    end
```

Comment:
Like adding in the additional_permitted_params, what is this decision_tree_class added in as a method for? Is this another example of black box behaviour?